app/src/common/shared/org/mozilla/vrbrowser/ui/widgets/prompts/AuthPromptWidget.java

```diff
@@ -0,0 +1,132 @@
+    private TextView mUsernameTextLabel;
+    private TextView mPasswordTextLabel;
+    private Button mOkButton;
+    private Button mCancelButton;
+    private GeckoSession.PromptDelegate.AuthCallback mCallback;
+
+    public AuthPromptWidget(Context aContext) {
+        super(aContext);
+        initialize(aContext);
+    }
+
+    public AuthPromptWidget(Context aContext, AttributeSet aAttrs) {
+        super(aContext, aAttrs);
+        initialize(aContext);
+    }
+
+    public AuthPromptWidget(Context aContext, AttributeSet aAttrs, int aDefStyle) {
+        super(aContext, aAttrs, aDefStyle);
+        initialize(aContext);
+    }
+
+    @Override
+    protected void initialize(Context aContext) {
+        super.initialize(aContext);
+
+        inflate(aContext, R.layout.prompt_auth, this);
+
+        mAudio = AudioEngine.fromContext(aContext);
+
+        mLayout = findViewById(R.id.layout);
+
+        mTitle = findViewById(R.id.textTitle);
+        mMessage = findViewById(R.id.textMessage);
+        mMessage.setMovementMethod(new ScrollingMovementMethod());
+        mUsernameText = findViewById(R.id.authUsername);
+        mUsernameText.setShowSoftInputOnFocus(false);
+        mUsernameTextLabel = findViewById(R.id.authUsernameLabel);
+        mPasswordText = findViewById(R.id.authPassword);
+        mPasswordText.setShowSoftInputOnFocus(false);
+        mPasswordTextLabel = findViewById(R.id.authPasswordLabel);
+
+        mOkButton = findViewById(R.id.positiveButton);
+        mOkButton.setSoundEffectsEnabled(false);
+        mOkButton.setOnClickListener(view -> {
+            if (mAudio != null) {
+                mAudio.playSound(AudioEngine.Sound.CLICK);
+            }
+
+            if (mCallback != null) {
+                mCallback.confirm(mUsernameText.getText().toString(), mPasswordText.getText().toString());
+            }
+
+            hide(REMOVE_WIDGET);
+        });
+
+        mCancelButton = findViewById(R.id.negativeButton);
+        mCancelButton.setSoundEffectsEnabled(false);
+        mCancelButton.setOnClickListener(view -> {
+            if (mAudio != null) {
+                mAudio.playSound(AudioEngine.Sound.CLICK);
+            }
+
+            onDismiss();
+        });
+
+        Switch showPasswordSwitch = findViewById(R.id.showPasswordSwitch);
+        showPasswordSwitch.setOnCheckedChangeListener((buttonView, isChecked) -> {
+            if (isChecked) {
+                mPasswordText.setInputType(InputType.TYPE_CLASS_TEXT);
+            } else {
+                mPasswordText.setInputType(InputType.TYPE_CLASS_TEXT | InputType.TYPE_TEXT_VARIATION_PASSWORD);
+            }
+        });
+    }
+
+    @Override
+    protected void onDismiss() {
+        hide(REMOVE_WIDGET);
+
+        if (mCallback != null) {
+            mCallback.dismiss();
+        }
+    }
+
+    public void setAuthOptions(GeckoSession.PromptDelegate.AuthOptions aOptions, GeckoSession.PromptDelegate.AuthCallback aCallback) {
```

Comment:
It looks like Firefox doesn't warn:
Fennec:
![fennec](https://user-images.githubusercontent.com/3066/53917402-43f4fd80-4019-11e9-9ace-db6abdf44bea.png)
Firefox Desktop:
<img width="467" alt="firefox" src="https://user-images.githubusercontent.com/3066/53917403-43f4fd80-4019-11e9-91d6-819b5626ca90.png">
GeckoView Example:
![gve](https://user-images.githubusercontent.com/3066/53917404-448d9400-4019-11e9-82f5-01330dfd0246.png)
Chrome for Android doesn't:
![cr_android](https://user-images.githubusercontent.com/3066/53917399-43f4fd80-4019-11e9-8a7a-0e89249a63be.png)
But Chrome for desktop does:
<img width="455" alt="cr-desktop" src="https://user-images.githubusercontent.com/3066/53917400-43f4fd80-4019-11e9-8807-4320ce68fbef.png">
Safari on both iOS and macOS do
iOS:
![ios](https://user-images.githubusercontent.com/3066/53917405-448d9400-4019-11e9-88d2-c7c85022fbdc.png)
macOS:
<img width="442" alt="safari" src="https://user-images.githubusercontent.com/3066/53917406-448d9400-4019-11e9-93a1-73615fd64d5e.png">

I think it would be okay to buck the Mozilla trend and warn the user.